ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockCause.java

```diff
@@ -31,29 +31,20 @@ public enum BadBlockReason {
 
   private final BadBlockReason reason;
   private final String description;
-  private final Optional<Throwable> exception;
-
-  public static BadBlockCause fromProcessingError(final Throwable t) {
-    final String description = t.getLocalizedMessage();
-    return new BadBlockCause(
-        BadBlockReason.EXCEPTIONAL_BLOCK_PROCESSING, description, Optional.of(t));
-  }
 
   public static BadBlockCause fromBadAncestorBlock(final Block badAncestor) {
     final String description =
         String.format("Descends from bad block %s", badAncestor.toLogString());
-    return new BadBlockCause(BadBlockReason.DESCENDS_FROM_BAD_BLOCK, description, Optional.empty());
+    return new BadBlockCause(BadBlockReason.DESCENDS_FROM_BAD_BLOCK, description);
   }
 
   public static BadBlockCause fromValidationFailure(final String failureMessage) {
-    return new BadBlockCause(
-        BadBlockReason.SPEC_VALIDATION_FAILURE, failureMessage, Optional.empty());
+    return new BadBlockCause(BadBlockReason.SPEC_VALIDATION_FAILURE, failureMessage);
   }
 
-  private BadBlockCause(BadBlockReason reason, String description, Optional<Throwable> exception) {
+  private BadBlockCause(final BadBlockReason reason, final String description) {
```

Comment:
I'm not sure why the missing `final`'s here only caused a compilation error / failed unit tests ([here](https://github.com/hyperledger/besu/actions/runs/8159257090/job/22303389097?pr=6678) and [here](https://github.com/hyperledger/besu/actions/runs/8159257090/job/22303387743?pr=6678)) when I pushed the toString change?

Reply:
Added an issue to investigate [here](https://github.com/hyperledger/besu/issues/6685).